google-cloud-util/google-cloud-compat-checker/src/main/java/com/google/cloud/compatchecker/GoogleCloudCompatChecker.java

```diff
@@ -43,30 +44,96 @@ protected void logProperty(String s, String s1) {
   }
 
   public static void main(String[] args) {
+    Boolean checkResult = check();
+    if (checkResult != null && !checkResult) {
+      System.exit(1);
```

Comment:
This is `main`, throwing exceptions is not helpful. We already have meaningful messages printed in `check()`. This is indeed meant to be run like a script, but it has to be in java to make the right calls and version checks.